### PR TITLE
CASMCMS-8679: Add dkms support for image customization

### PR DESCRIPTION
**Pull request by rbak-hpe**
## Summary and Scope

Adds the ability to specify configuration layers that require dkms to be enabled in the target image.

## Issues and Related PRs

* Resolves CASMCMS-8679

### Tested on:

  * Mug

### Test description:

Tested image customization sessions with different configs and combinations of layers with the dkms parameter.

## Risks and Mitigations

None

## Pull Request Checklist

- [X] Version number(s) incremented, if applicable
- [X] Copyrights updated
- [X] License file intact
- [X] Target branch correct
- [X] CHANGELOG.md updated
- [X] Testing is appropriate and complete, if applicable
- [X] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by dlaine-hpe — APPROVED** [review]

**Comment by deantroe** [comment]
@rbak-hpe we will want the SAT folks to be aware of this as we have canned SAT bootprep input files that will need to support this.  Pulling in @haasken-hpe to get the ball rolling.

**Comment by rbak-hpe** [comment]
I've opened CRAYSAT-1741 to track to SAT changes, and informed the SAT folks.

File: api/openapi.yaml [hunk 1]
```diff
@@ -650,6 +650,17 @@ components:
...
+        specialParameters:
+          type: object
+          description: |
+            Optional parameters that do not affect the configuration content or are only used in
+            special circumstances.
+          properties:
+            imsRequireDkms:
+              type: boolean
+              description: |
+                If true, any image customization sessions that use this configuration will enable dkms in IMS.
```
Comment (mharding-hpe) [inline api/openapi.yaml:662]:
```suggestion
                If true, any image customization sessions that use this configuration will enable DKMS in IMS.
```

**Review by mharding-hpe — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 - Quadrupled the size the uwsgi buffer for the API.
 
+### Added
+- Added a new configuration parameter for enabling dkms in ims
```
Comment (mharding-hpe) [inline CHANGELOG.md:12]:
```suggestion
- Added a new configuration parameter for enabling DKMS in IMS
```

File: api/openapi.yaml [hunk 3]
```diff
@@ -650,6 +650,17 @@ components:
             The configuration branch to use.  This will automatically set commit to master on the branch
             when the configuration is added.
           pattern: '^[^\s;]*$'
+        specialParameters:
+          type: object
+          description: |
+            Optional parameters that do not affect the configuration content or are only used in
+            special circumstances.
+          properties:
+            imsRequireDkms:
```
Comment (haasken-hpe) [inline api/openapi.yaml:659]: Does this default to `false`?

Comment (rbak-hpe) [reply api/openapi.yaml:659]: Not setting it is the same as not specifying the parameter when creating an IMS customization session.  Currently I believe that behavior is to default to false in IMS, but CFS does not have a default value.

Comment (mharding-hpe) [reply api/openapi.yaml:659]: In that case, maybe the description should be more along the lines of "If specified, the <whatever> parameter will be specified to the IMS session with the specified value". That makes it clear that setting it here just determines what value is passed into IMS, and IMS can use it as it pleases. Maybe also mention that if this is not specified, the parameter is not specified for the IMS session.

**Pull request merged**